apps/api/src/app/notification-template/e2e/create-notification-templates.e2e.ts

```diff
@@ -387,6 +417,210 @@ describe('Create Notification template - /notification-templates (POST)', async
     expect(result.credentials.senderName).to.equal('senderName');
   });
 
+  it('should filter in app seen/read step', async function () {
+    const firstStepUuid = uuid();
+    const template = await session.createTemplate({
+      steps: [
+        {
+          type: StepTypeEnum.IN_APP,
+          content: 'Not Delayed {{customVar}}' as string,
+          uuid: firstStepUuid,
+        },
+        {
+          type: StepTypeEnum.DELAY,
+          content: '',
+          metadata: {
+            unit: DigestUnitEnum.MINUTES,
+            amount: 5,
+            type: DelayTypeEnum.REGULAR,
+          },
+        },
+        {
+          type: StepTypeEnum.IN_APP,
+          content: 'Hello world {{customVar}}' as string,
+          filters: [
+            {
+              isNegated: false,
+              type: 'GROUP',
+              value: 'AND',
+              children: [
+                {
+                  on: FilterPartTypeEnum.PREVIOUS_STEP,
+                  stepType: PreviousStepTypeEnum.READ,
+                  step: firstStepUuid,
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    });
+
+    await axiosInstance.post(
+      `${session.serverUrl}/v1/events/trigger`,
+      {
+        name: template.triggers[0].identifier,
+        to: [subscriber.subscriberId],
+        payload: {
+          customVar: 'Testing of User Name',
+        },
+      },
+      {
+        headers: {
+          authorization: `ApiKey ${session.apiKey}`,
+        },
+      }
```

Comment:
Because we are using the Trigger event endpoint to test this, I find it weird to live in the create notification templates E2E. Even if the feature is related to notification creation. That also explains why you had to add so many dependencies to this file to be able to run the test. 
I particularly move these tests to the `trigger-event.e2e.ts` test file and create a dedicated `describe` for this feature.